workflows/requests.go

```diff
@@ -1,20 +1,36 @@
 package workflows
 
-type WorkflowReference struct {
-	InstanceID string `json:"instance_id"`
+// StartRequest is the struct describing a start workflow request.
+type StartRequest struct {
+	InstanceID    string            `json:"instanceID"`
+	Options       map[string]string `json:"options"`
+	WorkflowName  string            `json:"workflowName"`
+	WorkflowInput []byte            `json:"workflowInput"`
 }
 
-// StartRequest is the object describing a Start Workflow request.
-type StartRequest struct {
-	Options      map[string]string `json:"workflow_options"`
-	InstanceID   string            `json:"workflow_reference"`
-	WorkflowName string            `json:"function_name"`
-	Input        interface{}       `json:"input"`
+// GetRequest is the struct describing a get workflow state request.
+type GetRequest struct {
+	InstanceID string `json:"instanceID"`
+}
+
+// TerminateRequest is the struct describing a terminate workflow request.
+type TerminateRequest struct {
+	InstanceID string `json:"instanceID"`
 }
 
-// RaiseEventRequest is the object describing a Raise Event request.
+// RaiseEventRequest is the struct describing a raise workflow event request.
 type RaiseEventRequest struct {
-	InstanceID string `json:"workflow_reference"`
-	EventName  string `json:"event_name"`
-	Input      []byte `json:"input"`
+	InstanceID string `json:"instanceID"`
+	EventName  string `json:"name"`
+	EventData  []byte `json:"data"`
```

Comment:
Given the comment "We assume all inputs are JSON values" please consider using `json.RawValue` here. Two reasons:

First: It allows people to put valid JSON in there, but Go won't unmarshal it. So they don't need to encode it as string, they can pass valid JSON. For example:

```json
{
    "instanceID": "foo",
    "name": "bar",
    "data": {
        "some": ["random", "stuff", 42]
    }
}
```

Using the above with `[]byte` would fail because "data" is not a base64-encoded string. Users would have to input JSON as a base64-encoded string instead, which is not as nice.

Second, while json.RawMessage isn't parsed, it's still validated when doing json.Unmarshal to make sure it's valid JSON (so parentheses are closed, etc)

Reply:
Thanks for the review! This is actually really good info for me. However, the comment you're referring to is strictly for the Temporal component. It wasn't my intention that we assume this is the case for all workflow components.
One thing I haven't looked at closely, however, is how this impacts the HTTP API in `dapr/dapr`. I may revisit this if I think it makes the API too unweildy.

Reply:
Do you think it may be a fair assumption to make all components accept JSON?

Yes, it would be a change in the runtime too. I would personally welcome this as a good step however :)

Reply:
After spending some time on the HTTP API handler, I feel that my use of `[]byte` here is correct.

The plan is to take the entire HTTP request payload as the event data without making any assumptions about its format. It could be JSON, it could be plain text, it could be binary data - the API doesn't care. The other fields you mentioned in your example (`instanceID` and `name`) will come from the request URL.

This means that the caller can send any payload they want in the HTTP request and don't have to do any encoding on their side to satisfy our API layer. They just need to make sure that any data they send matches what the component implementation expects. Only the component will attempt to decode the event data.

This is the same for my changes to the `StartRequest` type, which also makes no assumptions about the format of the workflow input.

Reply:
Ok, that makes sense then and it's fine with me